File: recipe_scraper/schema.py
from uuid import UUID
from sqlmodel import Field, SQLModel


class Recipe(SQLModel, table=True):
    id: UUID = Field(primary_key=True)
    name: str | None = None
    description: str | None = None
    serving_size_people: int | None = None
    serving_size_is_pieces: bool = False
    time_estimate: int | None = None
    time_estimate_total: int | None = None
    difficulty_level: int | None = None
    image_id: str | None = None
    video_id: str | None = None
    number_of_ratings: int | None = None
    average_rating: float | None = None


class Section(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    recipe_id: UUID = Field(foreign_key="recipe.id")
    name: str | None = None
    sort_index: int | None = None


class Step(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    recipe_id: UUID = Field(foreign_key="recipe.id")
    section_id: int = Field(foreign_key="section.id")
    description: str | None = None
    sort_index: int | None = None
    tip: str | None = None
    image_id: str | None = None


class Ingredient(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    recipe_id: UUID = Field(foreign_key="recipe.id")
    section_id: int = Field(foreign_key="section.id")
    name: str | None = None
    sort_index: int | None = None
    in_pantry: bool = False
    amount: float | None = None
    unit: str | None = None
    measurement_value: float | None = None
    measurement_type: str | None = None
    additional_info: str | None = None

File: forkast-app/src/db/schema.ts
import {
    boolean,
    date,
    integer,
    pgEnum,
    pgTable,
    real,
    serial,
    text,
    uuid,
    varchar
} from 'drizzle-orm/pg-core';

export const mealEnum = pgEnum('meal', ['breakfast', 'lunch', 'dinner', 'supper', 'snack']);

export const recipe = pgTable('recipe', {
    id: uuid().primaryKey(),
    name: varchar({ length: 255 }),
    description: text(),
    servingSizePeople: integer('serving_size_people'),
    servingSizeIsPieces: boolean('serving_size_is_pieces').default(false),
    timeEstimate: integer('time_estimate'),
    timeEstimateTotal: integer('time_estimate_total'),
    difficultyLevel: integer('difficulty_level'),
    imageId: varchar('image_id', { length: 255 }),
    videoId: varchar('video_id', { length: 255 }),
    numberOfRatings: integer('number_of_ratings'),
    averageRating: real('average_rating')
});

export const section = pgTable('section', {
    id: serial().primaryKey(),
    recipeId: uuid('recipe_id').references(() => recipe.id),
    name: varchar({ length: 255 }),
    sortIndex: integer('sort_index')
});

export const step = pgTable('step', {
    id: serial().primaryKey(),
    recipeId: uuid('recipe_id').references(() => recipe.id),
    sectionId: integer('section_id').references(() => section.id),
    description: text(),
    sortIndex: integer('sort_index'),
    tip: text(),
    imageId: varchar('image_id', { length: 255 })
});

export const ingredient = pgTable('ingredient', {
    id: serial().primaryKey(),
    recipeId: uuid('recipe_id').references(() => recipe.id),
    sectionId: integer('section_id').references(() => section.id),
    name: varchar({ length: 255 }),
    sortIndex: integer('sort_index'),
    inPantry: boolean('in_pantry').default(false),
    amount: real(),
    unit: varchar({ length: 100 }),
    measurementValue: real('measurement_value'),
    measurementType: varchar('measurement_type', { length: 100 }),
    additionalInfo: text('additional_info')
});

export const mealPlanEntry = pgTable('meal_plan_entry', {
    id: serial().primaryKey(),
    recipeId: uuid('recipe_id').references(() => recipe.id),
    date: date().notNull(),
    meal: mealEnum().notNull(),
    notes: text(),
    servings: integer().default(2)
});

export const pantryItem = pgTable('pantry_item', {
    id: serial().primaryKey(),
    name: varchar({ length: 255 }).notNull(),
    amount: real(),
    unit: varchar({ length: 100 }),
    notes: text()
});
